share/cmake/macros/GetPythonPreCommand.cmake

```
@@ -42,15 +43,20 @@ macro(get_python_pre_command)
 
         # Build path list
         set(_WIN_PATHS 
-            ${_PYD_PATH} 
-            "${PROJECT_SOURCE_DIR}\\share\\docs"
+            ${_PYD_PATH}
+            "${_WIN_SOURCE_DIR}\\share\\docs"
         )
         # Include optional paths from macro arguments
         foreach(_PATH ${ARGN})
             file(TO_NATIVE_PATH ${_PATH} _WIN_PATH)
             list(APPEND _WIN_PATHS ${_WIN_PATH})
         endforeach()
-        list(APPEND _WIN_PATHS "%PYTHONPATH%")
+
+        # Double % to escape as the cmd.exe will eat one level of %. That
+        # results in an empty path in the ENV variable and Python fails to
+        # convert that to absolute path. Possibly due to
+        # https://www.cve.news/cve-2023-41105/
```

My comment was not accurate here. Please update
```suggestion
        # Double % to escape as an intermediate cmd.exe step causes early
        # expansion. That results in an empty path in the ENV variable and
        # Python 3.11+ fails to convert that to absolute path. Possibly due to
        # https://www.cve.news/cve-2023-41105/
```